lightning-invoice/src/utils.rs

```diff
@@ -126,12 +209,78 @@ where
 	let duration = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH)
 		.expect("for the foreseeable future this shouldn't happen");
 	create_invoice_from_channelmanager_and_duration_since_epoch(
-		channelmanager,
-		keys_manager,
-		network,
-		amt_msat,
-		description,
-		duration
+		channelmanager, keys_manager, network, amt_msat, description, duration
+	)
+}
+
+#[cfg(feature = "std")]
+/// Utility to construct an invoice. Generally, unless you want to do something like a custom
+/// cltv_expiry, this is what you should be using to create an invoice. The reason being, this
+/// method stores the invoice's payment secret and preimage in `ChannelManager`, so (a) the user
+/// doesn't have to store preimage/payment secret information and (b) `ChannelManager` can verify
+/// that the payment secret is valid when the invoice is paid.
+/// Use this variant if you want to pass the `description_hash` to the invoice.
+pub fn create_invoice_from_channelmanager_with_description_hash<
+	Signer: Sign,
```

Comment:
Same here - we're now 21 LoC to list 5 arguments, with two or three of these suddenly even large terminals have no context for what's going on..this is a great example of why rustfmt is unreadable. Can you condense these somewhat (I know some of the existing code was this way)